snippets/snippets.json

```diff
@@ -8,5 +8,55 @@
 		],
 		"description": "For Loop",
 		"scope": "source.julia"
+	},
+	"While Loop": {
+		"prefix": "while",
+		"body": [
+			"while ($1)",
+			"\t$2",
+			"end"
+		],
+		"description": "While loop",
+		"scope": "source.julia"
+	},
+	"Type Declaration": {
+		"prefix": "type",
+		"body": [
+			"type $1",
+			"\t$2",
+			"end"
+		],
+		"description": "Type declaration",
+		"scope": "source.julia"
+	},
+	"Immutable Declaration": {
+		"prefix": "immutable",
+		"body": [
+			"type $1",
+			"\t$2",
+			"end"
+		],
+		"description": "Immutable declaration",
+		"scope": "source.julia"
+	},
+	"Function Declaration": {
+		"prefix": "function",
+		"body": [
+			"function ($1::$2)",
```

Comment:
I don't think this should be the default, because a) there might be functions with no arguments and b) it is generally better in julia to actually not type function arguments.